jwst/assign_wcs/nirspec.py

```diff
@@ -662,31 +637,66 @@ def get_open_msa_slits(msa_file, msa_metadata_id, dither_position,
             quadrant = slitlets_sid[0]['shutter_quadrant']
             ycen = j
             xcen = slitlets_sid[0]['shutter_row']  # grab the first as they are all the same
+            shutter_id = xcen + (ycen - 1) * 365  # shutter numbers in MSA file are 1-indexed
+
+            # Background slits all have source_id=0 in the msa_file,
+            # so assign a unique id based on the slitlet_id
+            source_id = slitlet_id
+
+            # Hardwire the source info for background slits, because there's
+            # no source info for them in the msa_file
             source_xpos = 0.5
             source_ypos = 0.5
```

Comment:
@melanieclarke I believe this retains the fix you recently made, by initializing the x/ypos to 0.5 for background (and virtual) slits, which then later gets modified to 0.0.